langchain/src/vectorstores/prisma.ts

```diff
@@ -245,19 +286,38 @@ export class PrismaVectorStore<
     return results.map((result) => result[0]);
   }
 
+  async similaritySearchWithScore(
+    query: string,
+    k?: number,
+    filter?: TFilterModel
+  ) {
+    return super.similaritySearchWithScore(query, k, filter);
+  }
+
   async similaritySearchVectorWithScore(
     query: number[],
-    k: number
+    k: number,
+    filter?: TFilterModel
   ): Promise<[Document<SimilarityModel<TModel, TSelectModel>>, number][]> {
     const vectorQuery = `[${query.join(",")}]`;
     const articles = await this.db.$queryRaw<
       Array<SimilarityModel<TModel, TSelectModel>>
-    >`
-      SELECT ${this.selectSql}, ${this.vectorColumnSql} <=> ${vectorQuery}::vector as "_distance" 
-      FROM ${this.tableSql}
-      ORDER BY "_distance" ASC
-      LIMIT ${k};
-    `;
+    >(
+      this.Prisma.join(
+        [
+          this.Prisma.sql`
+            SELECT ${this.selectSql}, ${this.vectorColumnSql} <=> ${vectorQuery}::vector as "_distance"
```

Comment:
This is ok because everything here is generated in from a config file right or a vector array right? 

I still think it would be nice to parameterize things for best practices.

Reply:
Or does `this.Prisma.sql` sanitize?

Reply:
Base on the [doc](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#sql-injection):
> $executeRaw can only run one query at a time. You cannot append a second query - for example, adding DROP bobby_tables to the end of an ALTER.

Although it didn't mention queryRaw, but i found a [related issue](https://github.com/prisma/docs/issues/524) says:

> queryRaw(sql) and executeRaw(sql) will only be executing a single statement, which means some injections won't work (the one with multiple queries trying to drop tables for example) but some will (unveiling additional data using UNION for example) 

So I don't think this will be a problem.

Reply:
Yep! Parameters are by default sanitised unless they are wrapped in a `Prisma.raw` call. 

There are some dynamic fields, which cannot be parametrised (such as table names and column names). Moved these `Prisma.raw` calls closer to the `executeRaw` and `queryRaw` invocations alongside a note in docs to warn users.

Reply:
Ok. Using a single statement helps but I'm sure there's still sneaky issues and damage some bad metadata on a loaded document could do even in a single query, so please escape here: 

https://github.com/hwchase17/langchainjs/pull/1234/files#diff-9bc13719b32d4ab3d65a0d4f561ae4427f1c746ad7d140e30c0005b4e0d91018R271

Raw SQL from user provided fields is also really not ideal but at least it'll be the user's own fault if something happens there.